streams/src/main/java/org/apache/kafka/streams/kstream/internals/metrics/Sensors.java

```diff
@@ -38,13 +45,89 @@ public static Sensor lateRecordDropSensor(final InternalProcessorContext context
         );
         StreamsMetricsImpl.addInvocationRateAndCount(
             sensor,
-            "stream-processor-node-metrics",
+            PROCESSOR_NODE_METRICS_GROUP,
             metrics.tagMap("task-id", context.taskId().toString(), "processor-node-id", context.currentNode().name()),
             "late-record-drop"
         );
         return sensor;
     }
 
+    public static Sensor suppressionSensor(final InternalProcessorContext context) {
+        final StreamsMetricsImpl metrics = context.metrics();
+        final String operation = "intermediate-result-suppression";
+
+        final Sensor sensor = metrics.nodeLevelSensor(
+            context.taskId().toString(),
+            context.currentNode().name(),
+            operation,
+            Sensor.RecordingLevel.DEBUG
+        );
+
+        final Map<String, String> tags = metrics.tagMap(
+            "task-id", context.taskId().toString(),
+            "processor-node-id", context.currentNode().name()
+        );
+
+        sensor.add(
+            new MetricName(
+                operation + "-rate",
+                PROCESSOR_NODE_METRICS_GROUP,
+                "The average number of occurrence of " + operation + " operation per second.",
+                tags
+            ),
+            new Rate(TimeUnit.SECONDS, new Sum())
+        );
+        sensor.add(
+            new MetricName(
+                operation + "-total",
+                PROCESSOR_NODE_METRICS_GROUP,
+                "The total number of occurrence of " + operation + " operations.",
+                tags
+            ),
+            new Total()
+        );
+
+        return sensor;
+    }
+
+    public static Sensor suppressionEmitSensor(final InternalProcessorContext context) {
```

Comment:
This sensor is different than that proposed in the kip. During implementation, I noticed a weird asymmetry in which the processor node would measure the suppression, but the buffer would measure the "eviction" aka "emission" aka "forwarding".

I'm proposing to update the KIP to make these two measurements symmetrical.

Reply:
I didn't think of that before, but now that you mention it, the change makes sense to me.